vertical-stepped-line/script.js

```diff
@@ -1,109 +1,87 @@
-var annotations = [
+const annotations = [
   {
     value: -100,
     year: '2013/01',
     path: 'M93,-26L93,36',
-    text: 'Arsenal buys Alexis Sánchez',
+    text: 'Arsenal buy Alexis Sánchez',
     textOffset: [-25, -35],
   },
 ];
 
 // The config object passed by draw() gives us a width and height
-var config = { width: 600, height: 450 };
-var margin = { top: 30, right: 60, bottom: 50, left: 40 },
+const config = { width: 600, height: 450 };
+const margin = { top: 30, right: 60, bottom: 0, left: 40 },
   width = config.width - margin.left - margin.right,
   height = config.height - margin.top - margin.bottom;
 
 // Clean up SVG container before drawing
 d3.select('#times-vertical-line').html('');
 
-var svg = d3
-  .select('#times-vertical-line')
-  .attr('width', width)
-  .attr('height', height);
+const svg = d3.select('#times-vertical-line').at({
+  width: width,
+  height: height,
+});
 
 // Date parser
-var parseTime = d3.timeParse('%Y/%m');
+const parseTime = d3.timeParse('%Y/%m');
 
 // Scales
-var x = d3.scaleLinear().range([0, width]);
-var y = d3.scaleTime().range([height, 0]);
+const x = d3.scaleLinear().range([0, width]);
+const y = d3.scaleTime().range([height, 0]);
 
 // Line declaration
-var line = d3
+const line = d3
   .line()
-  .x(function(d) {
-    return x(d.value);
-  })
-  .y(function(d) {
-    return y(d.year);
-  })
+  .x(d => x(d.value))
+  .y(d => y(d.year))
   .curve(d3.curveStepAfter);
 
 // g is our container
-var g = svg
-  .append('g')
-  .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+const g = svg.append('g').translate([margin.left + 10, margin.top + 10]);
 
 d3.json('data.json', function(err, dataset) {
-  dataset.forEach(function(d) {
+  dataset.forEach(d => {
     d.year = parseTime(d.year);
     d.value = +d.value;
```

Comment:
I still hate this syntax (damnit Bostock). I think for newcomers it's totally baffling what it does, `parseInt` or `parseFloat` is so much clearer